users/models.py

```diff
@@ -55,14 +65,97 @@ def can_manage_event_groups_in_project(self, project):
+        self, child: Optional["Child"] = None
+    ) -> Union[models.QuerySet, list["FreeSpotNotificationSubscription"]]:
+        """
+        Get all the user's subscriptions.
+        If a child argument is given, only the subscriptions that are linked
+        to that given child are returned.
+        """
+        from subscriptions.models import FreeSpotNotificationSubscription
+
+        return FreeSpotNotificationSubscription.objects.user_subscriptions(self, child)
+
+    def unsubscribe_all_notification_subscriptions(
+        self, child: Optional["Child"] = None
+    ):
+        """
+        Unsubscribe user's all notification subsriptions.
+        If a child argument is given, the unsubscribing is done only
+        to subscriptions that are linked to the given child that
+        the user is a guardian to.
+
+        NOTE: This function deletes the FreeSpotNotifications, which are linked to
+        a Child and Occurrence instances. **It should be noted that the current
+        model architecture allows that a child can have multiple guardians,
+        so unsubscribe can delete some notifications from other users as well.
+        However, the UI apps has never allowed more than 1 guardian for a child.**
+        """
+        subscriptions = self.get_subscriptions(child=child)
+        count = subscriptions.count()
+        subscriptions.delete()
+        logger.info(
+            f"User {self.uuid} unsubscribed all ({count}) "
+            "subscriptions they had active%s"
+            % (f" for child {child.id}." if child else ".")
+        )
+        return count
+
+    def create_subscriptions_management_auth_token(self):
+        """
+        Create a new token that has an expiration date
+        and can be used to as an authorization token to manage
+        the user's subscriptions to notifications.
+        """
+        from verification_tokens.models import VerificationToken
+
+        # NOTE: Should the email be left empty?
+        # The only value the email gives here is that it is denormalized to db
+        try:
+            email = self.guardian.email
+            if not email:
+                email = self.email
+        except Guardian.DoesNotExist:
+            email = self.email
+
+        # NOTE: Should this get_or_create never expiring tokens instead?
+        # It would be worse in security perspective, because 1 token would
+        # serve forever, but it would be better in usability, because
+        # then the token would be the same in every email every time, forever.
+        return VerificationToken.objects.create_token(
+            self,
+            self,
+            VerificationToken.VERIFICATION_TYPE_SUBSCRIPTIONS_AUTH,
+            email=email,
+            expiry_minutes=getattr(
+                settings, "SUBSCRIPTIONS_AUTH_TOKEN_VALID_MINUTES", 30 * 24 * 60
+            ),  # 30 days
+            token_length=getattr(settings, "SUBSCRIPTIONS_AUTH_TOKEN_LENGTH", 16),
```

Comment:
https://docs.python.org/3/library/secrets.html#secrets.token_urlsafe
> Return a random URL-safe text string, containing nbytes random bytes. The text is Base64 encoded, so on average each byte results in approximately 1.3 characters.

&

> The [secrets](https://docs.python.org/3/library/secrets.html#module-secrets) module is used for generating cryptographically strong random numbers suitable for managing data such as passwords, account authentication, security tokens, and related secrets.

So token_length 16 -> 16 random bytes -> 16*8 = 128 bits, that looks good IMHO security-wise.